application/ui/components/param-row.jsx

```diff
@@ -0,0 +1,221 @@
+        }
+
+        if (_.isBoolean(param.defaultValue)) {
+            defaultValue = (param.defaultValue) ? 'true' : 'false';
+        } else {
+            defaultValue = param.defaultValue;
+        }
+
+        description = (param.required ? '**Required**. ' : '') + param.description +
+        (defaultValue ? ' **Default:** ' + defaultValue : '');
+
+        // Add any configuration validation errors
+        errors = this.validateParamConfiguration(param);
+        if (errors.length > 0) {
+            description += "\n\n**API CONFIGURATION ERROR**\n";
+            for (var i in errors) {
+                description += "* " + errors[i] + "\n";
+            }
+        }
+
+        innerHtml = {
+            __html: marked(description)
+        };
+
+        if (!description) {
+            return null;
+        }
+
+        return (
+            <td colSpan={5} dangerouslySetInnerHTML={innerHtml}></td>
+        );
+    },
+
+    render : function()
+    {
+        var includeCheckboxComponent, nullCheckboxComponent, inlineParam = null, rowParam = null;
+        var description;
+        var component = this;
+
+        includeCheckboxComponent = (
+            <input
+                type      = "checkbox"
+                name      = {'include-' + component.props.param.name}
+                onChange  = {component.props.onInclude}
+                checked   = {component.props.isIncluded}
+            />
+        );
+        if (component.props.param.type !== 'file') {
+            nullCheckboxComponent = (
+                <input
+                    type = "checkbox"
+                    name = {'null-' + component.props.param.name}
+                    onChange  = {component.props.onNull}
+                    checked   = {component.props.isNull}
+                />
+            );
+        } else {
+            nullCheckboxComponent = ('N/A');
+        }
+
+        switch (component.props.param.type) {
+            // row-params
+            case 'hash':
+            case 'object':
+                if (!Params) {
+                    Params = require('./params');
+                }
+                rowParam = (
+                    <tr>
+                        <td colSpan={5}>
+                            <Params
+                                key={component.props.key}
+                                params={component.props.param.params}
+                                onChange={component.props.onChange}
+                            />
+                        </td>
+                    </tr>
+                );
+                break;
+            case 'array':
+                rowParam = (
+                    <tr>
+                        <td colSpan={5}>
+                            <ArrayObject
+                                key={component.props.key}
+                                param={component.props.param.param}
+                                onChange={component.props.onChange}
```

Comment:
For consistency's sake, could we align the props here? (Also, the above two multi-line components)